llvm/lib/CodeGen/PreISelIntrinsicLowering.cpp

```diff
@@ -160,10 +160,7 @@ static bool lowerObjCCall(Function &F, const char *NewFn,
     auto *CB = cast<CallBase>(U.getUser());
 
     if (CB->getCalledFunction() != &F) {
-      objcarc::ARCInstKind Kind = objcarc::getAttachedARCFunctionKind(CB);
-      (void)Kind;
-      assert((Kind == objcarc::ARCInstKind::RetainRV ||
-              Kind == objcarc::ARCInstKind::UnsafeClaimRV) &&
+      assert((*objcarc::getAttachedARCFunction(CB) == &F) &&
```

Comment:
I *think* you don't need to dereference the std::optional for the compare, below should work

```suggestion
      assert(objcarc::getAttachedARCFunction(CB) == &F &&
```